### ISSUE TITLE
[Feature Request]: Expand and Document the Lua API

### ISSUE THREAD

**Issue by scalyone** [diasurgical/devilutionX#7544]
### Feature Type

Quality of Life

### Describe

I would like to start creating mods using Lua but currently I'm not sure where to begin as there is little to no documentation. I'm also not sure how to go about finding the right areas to hook as I am very unfamiliar with C++.

**Comment by StephenCWills** [comment]
Obviously, we intend to do this. It's pretty much the primary focus of the 1.6.0 release.

Just about the most complicated thing you can make with Lua at this point is a clock in the corner.

![image](https://github.com/user-attachments/assets/c83bd909-14f1-4e54-8686-7db8eaf8cb2e)

This much is already documented on Discord, and also here: https://github.com/diasurgical/devilutionX/pull/6775#issuecomment-1793493004

**Comment by AJenbo** [comment]
Well you can also play sounds so you could make an alarm if you got really fancy
